osgp/platform/osgp-adapter-domain-smartmetering/src/test/java/org/opensmartgridplatform/adapter/domain/smartmetering/application/mapping/ActualPowerQualityResponseDataMapperTest.java

```diff
@@ -0,0 +1,67 @@
+/**
+ * Copyright 2021 Alliander N.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+package org.opensmartgridplatform.adapter.domain.smartmetering.application.mapping;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.math.BigDecimal;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+
+import org.junit.jupiter.api.Test;
+import org.opensmartgridplatform.domain.core.valueobjects.smartmetering.ActualPowerQualityData;
+import org.opensmartgridplatform.domain.core.valueobjects.smartmetering.ActualValue;
+import org.opensmartgridplatform.dto.valueobjects.smartmetering.ActualPowerQualityDataDto;
+import org.opensmartgridplatform.dto.valueobjects.smartmetering.ActualValueDto;
+import org.opensmartgridplatform.dto.valueobjects.smartmetering.CaptureObjectDto;
+
+public class ActualPowerQualityResponseDataMapperTest {
+
+    private final MonitoringMapper mapper = new MonitoringMapper();
+
+    private final static Class<?>[] EXPECTED_CLASS = new Class<?>[] { String.class, Date.class, BigDecimal.class,
+            Long.class };
+
+    @Test
+    public void testConvertActualPowerQualityResponse() {
+        final ActualPowerQualityDataDto responseDto = this.makeResponseDataDto();
+        final ActualPowerQualityData response = this.mapper
+                .map(responseDto, ActualPowerQualityData.class);
+        assertThat(response).isNotNull();
+
+        assertThat(response.getActualValues().size())
+                .withFailMessage("response object should return same number of profilentries")
+                .isEqualTo(EXPECTED_CLASS.length);
+        int i = 0;
+        for (final ActualValue actualValue : response.getActualValues()) {
+            final Class<?> clazz = actualValue.getValue().getClass();
+            assertThat(clazz).withFailMessage("the return class should be of the same type")
+                             .isEqualTo(EXPECTED_CLASS[i++]);
+        }
+    }
+
+    private ActualPowerQualityDataDto makeResponseDataDto() {
+        return new ActualPowerQualityDataDto(this.makeCaptureObjectDtos(),
+                this.makeActualValueDtos());
+    }
+
+    private List<CaptureObjectDto> makeCaptureObjectDtos() {
+        return new ArrayList<>();
+    }
+
+    private List<ActualValueDto> makeActualValueDtos() {
```

Comment:
I would rename ActualValue to something more specific. ActualValue can mean anything.

Reply:
All ActualValue classes and references renamed into PowerQualityValue